### PR TITLE
Move to camera_id

### PR DESCRIPTION
**Pull request by mterzo**
Renamed thread to camera to more clearly define what thread is.  Kept backwards compatability with
thread so that old configs can still be used.

This is from the discussion on #154 

I've tested this with using thread and camera in the config file.

My editor automatically strips empty white space at the end of line, I picked up a few of those changes.

### CODE REVIEW

**Comment by mterzo** [comment]
Oh, i need to update motion_guide.html

File: motion.c [hunk 1]
```diff
@@ -2761,27 +2761,29 @@ int main (int argc, char **argv)
          * option is used, so start at 1 then and 0 otherwise.
          */
         for (i = cnt_list[1] != NULL ? 1 : 0; cnt_list[i]; i++) {
-            /* If i is 0 it means no thread files and we then set the thread number to 1 */
-            cnt_list[i]->threadnr = i ? i : 1;
```
Comment (jogu) [inline motion.c]: If I've understood correctly, there are still other places we use threadnr, so we probably don't want to delete this line? ie. we still want to set threader  (Or we want to update the other places... not sure.)

File: camera1-dist.conf.in [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
 
 ###########################################################
 # Capture device options
 ############################################################
 
+#
+# Camera Id
+# Id used to label the camera when inserting data into SQL or saving the
+# camera image to disk.  This is better than using thread number so that there
+# always is a consistent labe
```
Comment (jogu) [inline camera1-dist.conf.in]: We probably don't need the last sentence here anymore, as thread number is hopefully no longer too visible to the user. (same comment applies to other 3 config files too!)

**Comment by mterzo** [comment]
This is a large change does github have a WIP status on PR?  I wanted to get it early so everyone can look at what I took out of the other PR.  Threadnr is through the code. I probably did need to leave that other line in there. Or threadnr needs to be removed completely. 

I agree on the docs.

**Comment by Mr-Dave** [comment]
I think of the WIP as our own personal repos.  I think we can start a issues and reference them from  there for discussion.  It also helps with the process because last week I was thinking of doing this revision from the other PR(along with the conf.d change) and the discussion in issues may be able to help making sure we are not duplicating efforts.  

The camera id is much better than thread id but we should keep the thread number inside the code for the moment since there is at least one component(SDL) that is dependent upon and actual thread number.  

Regarding the docs, lets do some testing with this and make sure we have all the code in place.  Then we open the issue on updating all the docs (don't forget the motion.1 manual file)

**Comment by Mr-Dave** [comment]
FYI:  I am cleaning this up a bit

**Comment by mterzo** [comment]
Ok.

**Pull request merged**